notebooks/predicting_queries/simple_intent_suggestion.ipynb

```diff
@@ -0,0 +1,570 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "acc_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Check whats worth conditioning on\n",
+    "acc_df.groupby('clean').size().reset_index(name='count').sort_values('count',ascending=False)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Tokenise search terms\n",
+    "acc_df['search_terms'] = acc_df['search_terms'].map(lambda x: x.split(','))\n",
+    "acc_df = acc_df.explode('search_terms')\n",
+    "acc_df['tokens'] = acc_df['search_terms'].map(lambda x: tokenizer.tokenize(x.translate(translator).split()))\n",
+    "acc_df['valid_tokens'] = acc_df['tokens'].map(lambda x: [token for token in x if token in service_terms])\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "acc_df = acc_df.explode('valid_tokens')\n",
+    "acc_df = acc_df.dropna()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "acc_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's try conditioning search terms on nationality\n",
+    "nationality = acc_df[acc_df['clean'].isin(['nationality-eu','nationality-uk','nationality-row'])]\n",
+    "nationality = nationality[['clean','valid_tokens']].groupby('clean').aggregate(lambda x: list(x))\n",
+    "nationality['valid_token_count'] = nationality['valid_tokens'].map(lambda x:Counter(x))\n",
+    "nationality = nationality.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nationality"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Bag of words for nationality search terms\n",
+    "nat_bow = pd.DataFrame.from_records(nationality['valid_token_count'],columns=service_dis.columns,index=nationality['clean'])                                  \n",
+    "nat_bow = nat_bow.replace(np.nan,0) + 1\n",
+    "nat_bow = nat_bow.divide(nat_bow.sum(1),0)\n",
+    "nat_bow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "idx = np.argsort(entropy(np.broadcast_to(nat_bow.loc['nationality-uk'].to_numpy(),(service_dis.shape[0],service_dis.shape[1])),service_dis,axis=1))[0:20]\n",
+    "service_bow.iloc[idx].index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "idx = np.argsort(entropy(np.broadcast_to(nat_bow.loc['nationality-eu'].to_numpy(),(service_dis.shape[0],service_dis.shape[1])),service_dis,axis=1))[0:10]\n",
```

Comment:
I think this would benefit from splitting out and using named arguments.
```suggestion
    "# get entropy of a distribution for given probability values\n",
    "ent = entropy(pk=np.broadcast_to(array=nat_bow.loc['nationality-eu'].to_numpy(),\n",
                                                                "shape=service_dis.shape),\n",
                             "qk=service_dis\n",
                             "axis=1)\n",
    "idx = np.argsort(a=ent)[0:10]\n",
```

Reply:
agreeeed